src/postselection.jl

```diff
@@ -145,14 +138,9 @@ function initByMsmt_2D(data,Anum,numSegments,numMeas_A,indMeas,selectMeas,select
   datamat = Dict()
   var_con = Dict()
   data_con = Dict()
-  ind=1
-  for kk = 1:num_data_meas
-    while ind in Anum
-      ind = ind+1
-    end
-    data_D[kk] = data[ind]
-    ind+=1
-  end
+
+  Dnum = setdiff(1:length(data), Anum)
+  [data_D[kk] = data[Dnum[kk]] for kk = 1:num_data_meas]
```

Comment:
Since you aren't assigning this array to anything, you should just write this as a simple for loop.